### PR TITLE
AOSL-5 handle camp fields

### PR DESCRIPTION
**Pull request by monishdeb**

### CODE REVIEW

File: CRM/Aoservicelisting/Form/ProviderApplicationConfirm.php [hunk 1]
```diff
@@ -46,46 +40,35 @@ public function buildQuickForm() {
...
-    $this->setDefaults($defaults);
-    foreach ($this->_elements as $element) {
-      if (strpos($element->getName(), '[') !== FALSE) {
-         $key = substr($element->getName(), 0, strpos($element->getName(), '['));
-         $arrayKey = substr($element->getName(), strpos($element->getName(), '[') + 1, -1);
-         $element->setValue($defaults[$key][$arrayKey]);
-      }
-      if (!$entryFound) {
-        unset($campValues[$count]);
+    // this part is to render camp fields
```
Comment (seamuslee001) [inline CRM/Aoservicelisting/Form/ProviderApplicationConfirm.php:59]: Nice

File: CRM/Aoservicelisting/Form/ProviderApplicationForm.php [hunk 2]
```diff
@@ -27,27 +21,22 @@ public function setDefaultValues() {
...
-        'contact_id_a' => $loggedInContactId,
-        'relationship_type_id' => 74,
-      ]);
-      if (!empty($relationship['count'])) {
-        $this->organizationId = $relationship['values'][$relationship['id']]['contact_id_b'];
-        $this->set('organizationId', $relationship['values'][$relationship['id']]['contact_id_b']);
+    if (!empty($this->_loggedInContactID)) {
+      if (!empty($this->organizationId)) {
+        // isn't it obvious that a person is only allowed to submit the application only after agreeing to the waiver it is a required field to proceed
+        $defaults['waiver_field'] = 1;
```
Comment (seamuslee001) [inline CRM/Aoservicelisting/Form/ProviderApplicationForm.php]: @monishdeb My thinking about not prepopulating this is that i feel like there should be a 'reaffirming' of the waiver but may appreicate @JoeMurray 's thoughts on this. 

To summarise, When someone is logged in goes to update their/their org's service listing should the waiver checkbox be checked by default, I had it as No, Monish's chode will check it by default

File: CRM/Aoservicelisting/Form/ProviderApplicationForm.php [hunk 3]
```diff
@@ -97,11 +86,11 @@ public function setDefaultValues() {
...
 
   public function buildQuickForm() {
-
     CRM_Core_Resources::singleton()->addStyleFile('biz.jmaconsulting.aoservicelisting', 'css/providerformstyle.css');
 
+    $attr = empty($this->organizationId) ? [] : ['readonly' => TRUE];
     $serviceListingOptions = [1 => E::ts('Individual'), 2 => E::ts('Organization')];
-    $listingTypeField = $this->addRadio('listing_type', E::ts('Type of Service Listing'), $serviceListingOptions);
+    $listingTypeField = $this->addRadio('listing_type', E::ts('Type of Service Listing'), $serviceListingOptions, $attr);
     $organizationNameField = $this->add('text', 'organization_name', E::ts('Organization Name'));
```
Comment (seamuslee001) [inline CRM/Aoservicelisting/Form/ProviderApplicationForm.php]: @monishdeb I think we need to apply the same attr to Org Name and Org Email

Comment (seamuslee001) [reply CRM/Aoservicelisting/Form/ProviderApplicationForm.php]: @JoeMurray I remember you mentioning on the ticket that in edit mode we should free org and individual name fields are you able to confirm which individual name fields we should be freezing?

File: CRM/Aoservicelisting/Form/ProviderApplicationForm.php (hunk 2 above)
Comment (monishdeb) [reply CRM/Aoservicelisting/Form/ProviderApplicationForm.php]: @seamuslee001 yeah I thought when a primary contact submits an application he/she has to check the waiver checkbox to proceed, as its required field. So when the loggedin user visits the edit page again. it should be checked as what is stored, isn't it?

Comment (monishdeb) [reply CRM/Aoservicelisting/Form/ProviderApplicationForm.php]: Note: The setting default is only on the edit page.

Comment (seamuslee001) [reply CRM/Aoservicelisting/Form/ProviderApplicationForm.php]: @monishdeb from scrum channel

"SL: 1) when updating a listing should the waiver field be prechecked (given they would have had to check it initially)
JM: 1) No, make them positively agree."

**Comment by monishdeb** [comment]
I am merging this PR after discussing with team. Will bring the remaining fixes in separate PR.

**Pull request merged**